src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java

```diff
@@ -92,15 +126,31 @@
 
 	/**
 	 * Memory to allocate for a Pod.
+	 *
+	 * @deprecated Use spring.cloud.deployer.kubernetes.limits.memory
 	 */
+	@Deprecated
 	private String memory = "512Mi";
 
 	/**
 	 * CPU to allocate for a Pod.
+	 *
+	 * @deprecated Use spring.cloud.deployer.kubernetes.limits.cpu
 	 */
+	@Deprecated
 	private String cpu = "500m";
 
 	/**
+	 * Memory and CPU limits (i.e. maximum needed values) to allocate for a Pod.
+	 */
+	private Resources limits = new Resources("500m", "512Mi");
+
+	/**
+	 * Memory and CPU requests (i.e. guaranteed needed values) to allocate for a Pod.
+	 */
+	private Resources requests = new Resources("500m", "512Mi");
```

Comment:
We shouldn't default this, only set this if explicitly requested. If someone set the mem limit to "128Mi" this would fail with: ```[StatusCause(field=spec.template.spec.containers[0].resources.limits, message=Invalid value: "128Mi": must be greater than or equal to memory request, reason=FieldValueInvalid, additionalProperties={})]```